Backend/app/alchemist/src/Alchemist/Utils.hs

```diff
@@ -1,9 +1,48 @@
 module Alchemist.Utils where
 
+import Data.List (nub)
+import Data.List.Split (split, splitWhen, whenElt)
 import Kernel.Prelude hiding (hPutStr)
 import System.IO
+import Text.Regex.TDFA ((=~))
 
 writeToFile :: FilePath -> String -> IO ()
 writeToFile filename content = do
   withFile filename WriteMode $ \handle_ -> do
     hPutStr handle_ content
+
+typeDelimiter :: String
+typeDelimiter = "() "
+
+makeTypeQualified :: String -> String
+makeTypeQualified str = concatMap replaceOrKeep (split (whenElt (`elem` typeDelimiter)) str)
+  where
+    replaceOrKeep :: String -> String
+    replaceOrKeep word = fromMaybe (if '.' `elem` word || word `elem` ["", ")", "(", " "] then word else error "Type not determined") (getQualifiedImport word)
+
+figureOutImports :: [String] -> [String]
+figureOutImports fieldTypes =
+  let extractWords = splitWhen (`elem` typeDelimiter) <$> fieldTypes
+   in filter (not . null) $ concatMap (map makeTypeQualified) (nub extractWords)
+
+findMatchingSqlType :: String -> Maybe String
+findMatchingSqlType haskellType =
+  case filter ((haskellType =~) . fst) defaultSQLTypes of
+    [] -> Nothing
+    ((_, sqlType) : _) -> Just sqlType
+
+defaultSQLTypes :: [(String, String)]
+defaultSQLTypes =
+  [ ("Text", "CHARACTER(36)"),
+    ("Id ", "Varchar(50)"),
+    ("TimeOfDay", "UTCTime")
+  ]
+
+getQualifiedImport :: String -> Maybe String
+getQualifiedImport = \case
+  "Text" -> Just "Data.Text.Text"
+  "Maybe" -> Just "Data.Maybe.Maybe"
```

Comment:
One Maybe is extra.

Reply:
![Screenshot 2023-11-23 at 4 30 09 PM](https://github.com/nammayatri/nammayatri/assets/46071327/e44865af-1765-4435-b53b-3a944a096e12)

Type will also be coverted to fulll name. 
Maybe is the constructor name defined in module Data.Maybe